### ISSUE TITLE
Extend support for user secrets and appsettings to use .jsonc

### ISSUE THREAD

**Issue by Nairda015** [dotnet/runtime#89487]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the solution you'd like

Hey 
There is an issue about this already but it is closed and I think approached in the wrong way. https://github.com/dotnet/aspnetcore/issues/13070

Instead of changing from JSON to JSONC, I propose to extend the supported extensions list for all appsettings and usersecrets.
I know JSON with comments works so the only reason to add this is to remove red underlines.

<img width="861" alt="image" src="https://github.com/dotnet/aspnetcore/assets/44712992/af8cea9d-f552-4dc8-ab02-2aa66a4326e1">

**Comment by gregsdennis** [comment]
@mitchdenny what is the runtime team supposed to do with this?  Parsing commented JSON is already supported via the [JsonCommentingHandling](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsoncommenthandling?view=netcore-3.1) enum.

**Comment by mitchdenny** [comment]
@gregsdennis: I think the bot mis filed it as a JSON issue, its actually a configuration feature request by @Nairda015. I was going to backlog it in aspnetcore to see whether we got more demand for it but then realized that the implementation probably exists  in the runtime repo rather than the aspnetcore repo.

**Comment by Nairda015** [comment]
Thank you @mitchdenny

**Comment by layomia** [comment]
> I'm not sure which IDE you are using (Rider?). I'm surprised it doesn't at least give you the option to suppress these warnings?

@Nairda015 -- re: @mitchdenny's question, which IDE are you using and can the warning be suppressed there? Seems like the way to go since configuration binding already [supports JSON comments](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-7.0#comments-in-appsettingsjson).

**Comment by Nairda015** [comment]
You are right I can suppress this errors in Rider but I'm not sure if this is the way how this problem should be approached by everyone
